File: webui/src/api/meta.ts
import { http } from "./index";

// TMDB 元数据接口
export interface Meta {
  chineseName: string;
  year: string;
  tmdbID: number;
  season: number;
  episodeTotalNum: number;
  airWeekday: number | null;
  posterURL: string;
  backdropURL: string;
  overview: string;
  genres: string;
}

const metaAPI = {
  // 搜索电视剧（番剧）
  searchTVs: async (name: string): Promise<Meta[]> => {
    return http.get(`/meta/tvs?name=${encodeURIComponent(name)}`);
  },

  // 搜索电影（剧场版）
  searchMovies: async (name: string): Promise<Meta[]> => {
    return http.get(`/meta/movies?name=${encodeURIComponent(name)}`);
  },

  // 根据TMDB ID获取电视剧元数据
  getTVMeta: async (tmdbID: number): Promise<Meta> => {
    return http.get(`/meta/tv/${tmdbID}`);
  },

  // 根据TMDB ID获取电影元数据
  getMovieMeta: async (tmdbID: number): Promise<Meta> => {
    return http.get(`/meta/movie/${tmdbID}`);
  },
};

export default metaAPI;

File: backend/internal/router/gin/meta.go
package gin

import (
	"net/http"
	"strconv"

	"github.com/gin-gonic/gin"
)

// SearchTV 搜索电视剧
// GET /apis/v1/meta/tvs?name=xxx
func (r *Router) SearchTVs(c *gin.Context) {
	name := c.Query("name")
	meta, err := r.metaParser.SearchTVs(c.Request.Context(), name)
	if err != nil {
		writeError(c, err)
		return
	}
	c.JSON(http.StatusOK, meta)
}

// SearchMovie 搜索电影
// GET /apis/v1/meta/movies?name=xxx
func (r *Router) SearchMovies(c *gin.Context) {
	name := c.Query("name")
	meta, err := r.metaParser.SearchMovies(c.Request.Context(), name)
	if err != nil {
		writeError(c, err)
		return
	}
	c.JSON(http.StatusOK, meta)
}

// GetTVMeta 根据TMDB ID获取电视剧元数据
// GET /apis/v1/meta/tv/:id
func (r *Router) GetTVMeta(c *gin.Context) {
	idStr := c.Param("id")
	id, err := strconv.Atoi(idStr)
	if err != nil {
		writeError(c, err)
		return
	}

	meta, err := r.metaParser.ParseTV(c.Request.Context(), id)
	if err != nil {
		writeError(c, err)
		return
	}
	c.JSON(http.StatusOK, meta)
}

// GetMovieMeta 根据TMDB ID获取电影元数据
// GET /apis/v1/meta/movie/:id
func (r *Router) GetMovieMeta(c *gin.Context) {
	idStr := c.Param("id")
	id, err := strconv.Atoi(idStr)
	if err != nil {
		writeError(c, err)
		return
	}

	meta, err := r.metaParser.ParseMovie(c.Request.Context(), id)
	if err != nil {
		writeError(c, err)
		return
	}
	c.JSON(http.StatusOK, meta)
}
